src/main/java/org/springframework/data/redis/connection/jedis/JedisConnectionFactory.java

```diff
@@ -776,22 +768,58 @@ protected Pool<Jedis> createRedisSentinelPool(RedisSentinelConfiguration config)
 	 * @since 1.4
 	 */
 	protected Pool<Jedis> createRedisPool() {
-		return new JedisPool(getPoolConfig(), new HostAndPort(getHostName(), getPort()), this.clientConfig);
+		return new JedisPool(getPoolConfig(), newHostAndPort(getHostName(), getPort()), this.clientConfig);
 	}
 
 	/**
-	 * Template method to create a {@link ClusterTopologyProvider} given {@link JedisCluster}. Creates
-	 * {@link JedisClusterTopologyProvider} by default.
-	 *
-	 * @param cluster the {@link JedisCluster}, must not be {@literal null}.
-	 * @return the {@link ClusterTopologyProvider}.
-	 * @see JedisClusterTopologyProvider
-	 * @see 2.2
+	 * @deprecated Use {@link #createClusterTopologyProvider(JedisCluster)} instead.
+	 * @since 2.2
 	 */
+	@Deprecated(since = "3.2")
 	protected ClusterTopologyProvider createTopologyProvider(JedisCluster cluster) {
 		return new JedisClusterTopologyProvider(cluster);
 	}
 
+	/**
+	 * Template method to create a {@link ClusterTopologyProvider} with the given {@link JedisCluster}.
+	 * <p>
+	 * Creates {@link JedisClusterTopologyProvider} by default.
+	 *
+	 * @param cluster reference to the configured {@link JedisCluster} used by the {@link ClusterTopologyProvider}
+	 * to interact with the Redis cluster; must not be {@literal null}.
+	 * @return a new {@link ClusterTopologyProvider}.
+	 * @see org.springframework.data.redis.connection.jedis.JedisClusterConnection.JedisClusterTopologyProvider
+	 * @see redis.clients.jedis.JedisCluster
+	 * @since 3.2
+	 */
+	protected ClusterTopologyProvider createClusterTopologyProvider(JedisCluster cluster) {
+		return createTopologyProvider(cluster);
+	}
+
+	/**
+	 * Create a new {@link ClusterCommandExecutor} with the given {@link JedisCluster} and {@link ClusterTopologyProvider}.
+	 * <p>
+	 * Creates a {@link JedisClusterNodeResourceProvider} by default.
+	 *
+	 * @param cluster reference to the configured {@link JedisCluster} and {@link ClusterTopologyProvider} used to
+	 * execute Redis commands across the cluster; must not be {@literal null}.
+	 * @param clusterTopologyProvider {@link ClusterTopologyProvider} used to gather information about the current
+	 * Redis cluster topology.
+	 * @return a new {@link ClusterCommandExecutor}.
+	 * @see org.springframework.data.redis.connection.ClusterTopologyProvider
+	 * @see redis.clients.jedis.JedisCluster
+	 * @since 3.2
+	 */
+	protected ClusterCommandExecutor createClusterCommandExecutor(JedisCluster cluster,
```

Comment:
No need to create duplicate `@see` references to types that are already present in the signature.

Reply:
Changed